graylog2-server/src/main/java/org/graylog/plugins/map/geoip/GeoIpDbFileChangeMonitorService.java

```diff
@@ -0,0 +1,182 @@
+/*
+ * Copyright (C) 2020 Graylog, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the Server Side Public License, version 1,
+ * as published by MongoDB, Inc.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * Server Side Public License for more details.
+ *
+ * You should have received a copy of the Server Side Public License
+ * along with this program. If not, see
+ * <http://www.mongodb.com/licensing/server-side-public-license>.
+ */
+package org.graylog.plugins.map.geoip;
+
+import com.google.common.eventbus.EventBus;
+import com.google.common.eventbus.Subscribe;
+import org.graylog.plugins.map.config.DatabaseType;
+import org.graylog.plugins.map.config.DatabaseVendorType;
+import org.graylog.plugins.map.config.GeoIpResolverConfig;
+import org.graylog2.cluster.ClusterConfigChangedEvent;
+import org.graylog2.events.ClusterEventBus;
+import org.graylog2.plugin.cluster.ClusterConfigService;
+import org.graylog2.plugin.system.NodeId;
+import org.graylog2.plugin.utilities.FileInfo;
+import org.graylog2.plugin.validate.ConfigValidationException;
+import org.graylog2.rest.resources.system.GeoIpResolverConfigValidator;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.inject.Inject;
+import javax.inject.Named;
+import javax.inject.Singleton;
+import java.nio.file.Paths;
+import java.time.Duration;
+import java.time.Instant;
+import java.util.EnumMap;
+import java.util.Map;
+import java.util.Objects;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+
+@Singleton
+public final class GeoIpDbFileChangeMonitorService {
+
+    private static final Logger LOG = LoggerFactory.getLogger(GeoIpDbFileChangeMonitorService.class.getSimpleName());
+
+    private ScheduledFuture<?> refreshTask;
+
+    private final ScheduledExecutorService scheduler;
+    private final GeoIpResolverConfigValidator geoIpResolverConfigValidator;
+
+    private final ClusterConfigService clusterConfigService;
+    private final ClusterEventBus clusterEventBus;
```

Comment:
```suggestion
    private final EventBus eventBus;
```
I am not certain, but _think_ the `EventBus` should be used here instead of the `ClusterEventBus`. I believe `ClusterEventBus` will broadcast the changed event to all Graylog nodes when a GeoIp the file is changed on just one node. I think we want the event to only be broadcast to all processors running on the same-node, since the file change only applies to the current node.

Reply:
Updated to use EventBus.